common-content/en/module/tools/programming-language-concepts/index.md

```diff
@@ -0,0 +1,60 @@
++++
+title = "Programming language concepts"
+headless = true
+time = 120
+facilitation = false
+emoji= "📖"
+objectives=[
+    "Describe what a variable is.",
+    "Describe how a variable relates to a memory location.",
+    "Give examples of common operators.",
+    "Describe the meaning of the `&&` (`and`), `||` (`or`), and `!` (`not`) operators.",
+    "Explain what happens when you call a function.",
+]
++++
+
+Read the learning objectives listed on this page: Bear in mind what you’re trying to achieve while reading this text. If a topic isn’t making much sense, and isn’t in the objectives, you can probably skip over it. If a topic is listed in the objectives, you should keep studying it until you are confident you’ve met the objective.
+
+{{<note type="Reading">}}
+Read chapter 9 of How Computers Work.
+
+Do every exercise listed in the chapter.
+
+You can skip the projects (though you're welcome to try any of them if you have time!).
+{{</note>}}
+
+Check you have achieved each learning objective listed on this page.
+
+{{<note type="Exercise">}}
+Write down an explanation of what a variable is. Make sure to use the term "memory location" in your explanation.
+{{</note>}}
+
+{{<note type="Exercise">}}
+Write down what operators you have used in a programming language.
+
+Next to each, write down whether it's a unary, binary, or ternary operator.
+
+Make sure you have at least one example of a unary operator, a binary operator, and a ternary operator.
+{{</note>}}
+
+{{<note type="Exercise">}}
+Write down what happens when you call a function.
+
+Make sure you include how the "next line of code to run" moves around, parameters/arguments, return values, and scope.
+{{</note>}}
+
+{{<note type="Exercise">}}
+Write down an explanation of what a class is, which could be understood by someone who's never written any code before.
```

Comment:
They will need much more guidance on this or they will get stuck for hours.  A lot of trainees have no practice in composing arguments and have never written an essay of any kind. 

Let's show them how to define a class in this way. 

```suggestion
Now write an explanation of what a class is, which could be understood by someone who's never written any code before. To help you do this, answer each of these sub questions and then put your answers together to form your explanation. 

1. How is a class like a Big Mac?
2. What is the difference between a burger template, like a Big Mac recipe, and the Big Mac you order for yourself in the store?
```

Reply:
Assign a Big Mac related energiser/demo thing in the following week.

Reply:
Similar to my comment above - there's several points here where trainees are instructed to "write down" something. Can we push them more explicitly towards introspection of their understanding?

I recognise that getting them to submit it for review is unlikely to go anywhere though. And perhaps we are making an assumption that trainees at this stage already know they need to do this?